wbooks-api/spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe BooksController, type: :controller do
+
+  describe "GET #index" do
+    it "returns http success" do
+      get :index
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "GET #show" do
+    it "returns http success" do
+      get :show
```

Comment:
A show with no id param should fail, too. Show should always take one ID and fail otherwise. And it'll help if you test:
- What happens with a correct ID (you get a book with X, Y and Z params, and you also get a code 200 as a response)
- What happens with an ID for a book that doesn't exist. It should fail (how?). This test will force you to handle failure in the controller

Reply:
Same!